multimodal/src/autogluon/multimodal/models/mmocr.py

```diff
@@ -0,0 +1,245 @@
+
+
+class MMOCRAutoModel(nn.Module):
+    """
+    Support MMOCR models.
+    Refer to https://github.com/open-mmlab/mmocr
+    """
+
+    def __init__(self, prefix: str, det_ckpt_name: str, recog_ckpt_name: str, kie_ckpt_name: str):
+        """
+        Load a pretrained ocr text detector from MMOCR.
+
+        Parameters
+        ----------
+        prefix
+            The prefix of the MMOCRAutoModel model.
+        det_ckpt_name
+            Name of the text detection checkpoint.
+        recog_ckpt_name
+            Name of the text recognition checkpoint.
+        kie_ckpt_name
+            Name of the key information extraction checkpoint.
+        """
+        super().__init__()
+
+        self.recog_model = None
+        if recog_ckpt_name != "None":
+            recog_config, self.recog_model = get_mmocr_config_and_model(recog_ckpt_name)
+            self.recog_model = revert_sync_batchnorm(self.recog_model)
+            self.recog_model.cfg = recog_config
+            self.model = self.recog_model
+            self.config = recog_config
+
+        self.det_model = None
+        if det_ckpt_name != "None":
+            det_config, self.det_model = get_mmocr_config_and_model(det_ckpt_name)
+            self.det_model = revert_sync_batchnorm(self.det_model)
+            self.det_model.cfg = det_config
+            self.model = self.det_model
+            self.config = det_config
+
+        # TODO
+        self.kie_model = None
+        if kie_ckpt_name != "None":
+            kie_config, self.kie_model = get_mmocr_config_and_model(kie_ckpt_name)
+
+        if self.det_model != None and self.recog_model != None:
+            recog_config.data.test.pipeline[0].type = "LoadImageFromNdarray"
+            recog_config.data.test.pipeline = replace_ImageToTensor(recog_config.data.test.pipeline)
+            self.recog_test_pipeline = Compose(recog_config.data.test.pipeline)
+
+        self.prefix = prefix
+
+    @property
+    def image_key(self):
+        return f"{self.prefix}_{IMAGE}"
+
+    @property
+    def image_valid_num_key(self):
+        return f"{self.prefix}_{IMAGE_VALID_NUM}"
+
+    @property
+    def label_key(self):
+        return f"{self.prefix}_{LABEL}"
+
+    @property
+    def image_column_prefix(self):
+        return f"{self.image_key}_{COLUMN}"
+
+    @property
+    def image_feature_dim(self):
+        return self.model.num_features
+
+    def forward(
+        self,
+        batch: dict,
+    ):
+        """
+        Parameters
+        ----------
+        batch
+            A dictionary containing the input mini-batch data.
+            We need to use the keys with the model prefix to index required data.
+
+        Returns
+        -------
+            A dictionary with bounding boxes.
+        """
+
+        data = batch[self.image_key]
+        # single image
+        if isinstance(data["img_metas"], List):
+            data["img_metas"] = [img_metas.data[0] for img_metas in data["img_metas"]]
```

Comment:
Data unpacks and to(device) shall be moved to collate function (maybe in next pr). For details see #2244.